ambari-infra-manager/src/main/java/org/apache/ambari/infra/job/SolrDAOBase.java

```diff
@@ -57,7 +61,10 @@ protected void delete(String deleteQueryText) {
   }
 
   protected CloudSolrClient createClient() {
-    CloudSolrClient client = new CloudSolrClient.Builder().withZkHost(zooKeeperConnectionString).build();
+    ConnectStringParser connectStringParser = new ConnectStringParser(zooKeeperConnectionString);
+    CloudSolrClient client = new CloudSolrClient.Builder(
+            connectStringParser.getServerAddresses().stream().map(InetSocketAddress::toString).collect(Collectors.toList()),
```

Comment:
I think extracting the result of this stream operation to a local variable would improve readability.